config/main.py

```diff
@@ -570,6 +570,102 @@ def is_ipaddress(val):
         return False
     return True
 
+def  interface_is_in_vlan(vlan_member_table, interface_name):
+    """ Check if an interface  is in a vlan """
+    for _,v in vlan_member_table.keys():
+        if v == interface_name:
+            return True
+
+    return False
+
+def  interface_is_in_portchannel(portchannel_member_table, interface_name):
+    """ Check if an interface is part of portchannel """
+    for _,v in portchannel_member_table.keys():
+        if v == interface_name:
+            return True
+
+    return False
+
+def interface_is_router_port(interface_table, interface_name):
+    """ Check if an interface has router config """
+    for entry in interface_table.keys():
+        if (interface_name == entry[0]):
+            return True
+
+    return False
+
+def interface_is_mirror_dst_port(config_db, interface_name):
+    """ Check if port is already configured as mirror destination port """
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+    for _,v in mirror_table.items():
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+def interface_has_mirror_config(mirror_table, interface_name):
+    """ Check if port is already configured with mirror config """
+    for _,v in mirror_table.items():
+        if 'src_port' in v and v['src_port'] == interface_name:
+            return True
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+#
+# Check if SPAN mirror-session config is valid.
+#
```

Comment:
Extra comment, can remove this now that the docstring is present.

Reply:
removed.